Clockwise/ICircuitBreakerStorage.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace Clockwise
+{
+    public interface ICircuitBreakerStorage : IDisposable
+    {
+        event EventHandler<CircuitBreakerStateDescriptor> CircuitBreakerStateChanged;
+        CircuitBreakerStateDescriptor StateDescriptor { get; }
+        void SetState(CircuitBreakerState newState, TimeSpan? expiry = null);
```

Comment:
If this is intended to perform network I/O it should be async and support cancelation.